ledger.go

```diff
@@ -271,6 +280,37 @@ func (l *Ledger) Snapshot() *avl.Tree {
 	return l.accounts.Snapshot()
 }
 
+func (l *Ledger) BroadcastingNop() bool {
+	l.broadcastNopsLock.Lock()
+	broadcastNops := l.broadcastNops
+	defer l.broadcastNopsLock.Unlock()
+
+	return broadcastNops
+}
+
+// WaitForConsensus blocks until the ledger reaches consensus.
```

Comment:
@iwasaki-kenta Need your thought on this. This function is basically added purely for testing purposes. What it does is it waits for a round to be finalized, or timeout if it takes too long. I'm not sure if it will impact the ledger in production, as it involves using a mutex. And it doesn't feel right to me at the moment.

Is there a better way to do this? Basically I just need a way to listen for a finalized round when testing. Otherwise either I'll just use `time.Sleep` (which makes the test run longer), or somehow poll the HTTP API to see if the round has increased.

Reply:
Nvm, I just polled `ledger.Round().Latest().Index` to see if the round increased, and it works quite well.